ast/src/analyzed/mod.rs

```diff
@@ -242,10 +242,14 @@ impl Symbol {
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// The "kind" of a symbol. In the future, this will be mostly
+/// replaced by its type.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
 pub enum SymbolKind {
     /// Fixed, witness or intermediate polynomial
     Poly(PolynomialType),
+    /// A constant value.
+    Constant(),
     /// Other symbol, could be a constant, depends on the type.
```

Comment:
```suggestion
    /// Other symbol
```
Can't be a constant anymore, right?

I think it would also be nice to list some examples of what it could be. I think array of numbers or lambda functions?